lib/rules/container-name-pattern/index.mjs

```diff
@@ -0,0 +1,118 @@
+import valueParser from 'postcss-value-parser';
+
+import { atRuleParamIndex, declarationValueIndex } from '../../utils/nodeFieldIndices.mjs';
+import { atRuleRegexes, propertyRegexes } from '../../utils/regexes.mjs';
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { basicKeywords } from '../../reference/keywords.mjs';
+import isStandardSyntaxAtRule from '../../utils/isStandardSyntaxAtRule.mjs';
+import isStandardSyntaxDeclaration from '../../utils/isStandardSyntaxDeclaration.mjs';
+import isStandardSyntaxValue from '../../utils/isStandardSyntaxValue.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'container-name-pattern';
+
+const messages = ruleMessages(ruleName, {
+	expected: (containerName, pattern) => `Expected "${containerName}" to match pattern "${pattern}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/container-name-pattern',
+};
+
+const KEYWORDS = new Set(['and', 'or', 'none', 'not']);
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: [isRegExp, isString],
+		});
+
+		if (!validOptions) return;
+
+		const regex = isString(primary) ? new RegExp(primary) : primary;
+
+		root.walkDecls(propertyRegexes.containerName, (decl) => {
+			if (!isStandardSyntaxDeclaration(decl)) return;
+
+			const parsedValue = valueParser(decl.value);
+
+			parsedValue.walk(({ sourceIndex, type, value }) => {
+				if (type !== 'word') return false;
+
+				if (basicKeywords.has(value.toLowerCase())) return;
+
+				if (!isStandardSyntaxValue(value)) return;
+
+				if (regex.test(value)) return;
+
+				complain(declarationValueIndex(decl) + sourceIndex, value, decl);
+			});
+		});
+
+		root.walkDecls(propertyRegexes.containerNameShorthand, (decl) => {
+			if (!isStandardSyntaxDeclaration(decl)) return;
+
+			const parsedValue = valueParser(decl.value);
+
+			parsedValue.walk((node) => {
+				let { sourceIndex, type, value } = node;
+
+				if (type !== 'word') return false;
+
+				if (node === parsedValue.nodes[parsedValue.nodes.length - 1]) return;
+
+				if (basicKeywords.has(value.toLowerCase())) return;
+
+				if (!isStandardSyntaxValue(value)) return;
+
+				if (regex.test(value)) return;
+
+				complain(declarationValueIndex(decl) + sourceIndex, value, decl);
+			});
+		});
```

Comment:
Let's recombine these into a single block that checks the `container` and `container-name` properties to remove the repetition and address the multiple container types test.

> This will involve updating the logic to check all the `word` nodes before `/` rather than just the first node.

There may be a more eloquent way of doing this, but this should work:

```js
let isContainerType = false;

parsedValue.walk((node) => {
	let { sourceIndex, type, value } = node;

	if (isContainerType) return;

	if (type === 'div' && value === '/') isContainerType = true;

	if (type !== 'word') return false;
```